Framework/Kernel/src/TimeROI.cpp

```diff
@@ -60,6 +60,7 @@ bool overlaps(const TimeInterval &left, const TimeInterval &right) {
 } // namespace
 
 const std::string TimeROI::NAME = "Kernel_TimeROI";
+const TimeROI TimeROI::INVALID_ROI{DateAndTime::GPS_EPOCH - DateAndTime::ONE_SECOND, DateAndTime::GPS_EPOCH};
```

Comment:
I have some beef with the chosen name `INVALID_ROI` because in my opinio it's a confusing name. I think we're using this ROI to indicate a mask that masks all times, so to me this is a _valid_ ROI and I'd rather call this as `EMPTY_ROI`. 
An ROI with returning `this->empty() == true` is an undefined or non-initialized ROI rather than an empty ROI. We're using "empty" to denote the underlying data structure is empty, but this concept doesn't apply to the ROI itself.
 I'd rather rename this method to `this->isUndefined()`. Then I would rename method `this->isValid()` as `this->isEmpty()`